File: src/template/ocaml/bfs.ml
let bfs start adj =
  let n = Array.length adj in
  let dist = Array.make n max_int in
  dist.(start) <- 0;
  let q = Queue.create () in
  Queue.add start q;
  while not (Queue.is_empty q) do
    let u = Queue.take q in
    List.iter (fun v ->
      if dist.(v) = max_int then begin
        dist.(v) <- dist.(u) + 1;
        Queue.add v q
      end) adj.(u)
  done;
  dist

File: src/template/julia/bfs.jl
function bfs(start::Int, adj::Vector{Vector{Int}})
    n = length(adj)
    dist = fill(typemax(Int), n)
    dist[start] = 0
    queue = [start]
    for u in queue
        for v in adj[u]
            if dist[v] == typemax(Int)
                dist[v] = dist[u] + 1
                push!(queue, v)
            end
        end
    end
    return dist
end